### DISCUSSION TITLE
How to solve 'cycle detected' problem with common module and JPA annotations

### DISCUSSION THREAD

**Discussion by cjacky475** [spring-projects/spring-modulith#557] [General]
Hello,

I am trying to move my project to use **Spring Modulith** but currently facing some issues with cycle dependencies:

1. What is the common idea when module needs to get data from the other module. Let's say there is **users** module and it needs to access its items from **items** module. Now in my example everything is sort of connected - each modules talks to each other via the provided service, but here the cyclic dependency occurs. So what are the best practices to solve this issue? Any decent examples and thorough explanation?
2. What about when I use JPA and **user** entity is associated with **ManyToMany** with **post** entity and post also contain user. Here is cycle dependency as well? This is also indication that modules are connected, but what is the practice to solve this then?

Thanks.

**Comment by shwetakawale** [comment]
Did you get any solution? 
there could be situation like
Module A (Order) depends on Module B (Inventory):
Whenever a new order is placed, Module A needs to call Module B to check if the required inventory is available.
Module B (Inventory) depends on Module A (Order):
Module B needs to be notified when an order is placed, updated, or canceled so it can adjust inventory levels accordingly.

but in spring modulith it leads to cyclic dependency, there could many such usecases, is there any way to solve this @odrotbohm
